app/src/main/java/com/hedvig/app/feature/marketpicker/MarketPickerViewModel.kt

```diff
@@ -18,43 +18,31 @@ import com.hedvig.app.shouldOverrideFeatureFlags
 import com.hedvig.app.util.apollo.defaultLocale
 import com.hedvig.app.util.extensions.getLanguage
 import com.hedvig.app.util.extensions.getMarket
-import com.hedvig.app.util.extensions.getStoredBoolean
 import kotlinx.coroutines.launch
 
 abstract class MarketPickerViewModel(private val context: Context) : ViewModel() {
     protected val _data = MutableLiveData<PickerState>()
     val data: LiveData<PickerState> = _data
     abstract fun uploadLanguage()
 
+    fun submitLanguageAndReload() {
+        uploadLanguage()
+        persistPickerState()
+        broadcastLocale()
+    }
+
     @SuppressLint("ApplySharedPref") // We want to apply this right away. It's important
-    fun save() {
+    private fun persistPickerState() {
         val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
-        var clean = false
-        data.value?.let { ps ->
-            clean = ps.market == context.getMarket() && ps.language == context.getLanguage()
-        }
-
         data.value?.let { data ->
             sharedPreferences.edit()
-                .putString(
-                    Market.MARKET_SHARED_PREF,
-                    data.market?.name
-                )
-                .commit()
-
-            sharedPreferences
-                .edit()
+                .putString(Market.MARKET_SHARED_PREF, data.market?.name)
```

Comment:
I like that we're batching this edit together - perhaps we could improve it even further by using `core-ktx`: https://developer.android.com/reference/kotlin/androidx/core/content/package-summary#(android.content.SharedPreferences).edit(kotlin.Boolean,%20kotlin.Function1)

which would make this something like this:


```kotlin
sharedPreferences.edit(commit=true) { // putString twice }
```

Reply:
Brilliant!